Add initial model that predicts maps in problem statements

VictorSuciu opened the pull request:
moved json reader from labeler.py to classifier.py since it's also needed for embedding.py. Just 
thought a more centralized location would be easier.

small bug fixes to labeler, embedder, classifier, and ModelData

Note: in problems.json, the last problem had a null value for the statement. I added the statement string manually.

Commit:
Now fully runs - accuracy around 83%
moved json reader from labeler.py to classifier.py since it's also needed for embedding.py. Just thought a more centralized location would be easier.
small bug fixes to labeler, embedder, classifier, and ModelData
Note: in problems.json, the last problem had a null value for the statement. I added the statement string manually.

VictorSuciu commented:
Whoops I didn't add .DS_Store and .vscode to gitignore

isabelgiang commented on src/parser.py:
```diff
@@ -63,9 +63,9 @@ def create_output_file():
             
 def main():
     print("Processing problem statements...")
-    process_statements("data/statements.txt")
```
Why is this file path being changed?

VictorSuciu replied:
Oh sorry my vs code environment defaults the working directory to the root folder instead of src, so on my end I have to specify src in the file path. I'll re commit with src removed

VictorSuciu replied:
I think it committed the fix. Let me know if it didn't work

Commit:
fixed file paths in parser.py
updated .gitignore
removed junk files specified in .gitignore

isabelgiang commented:
Looks like `.DS_Store` is still included in the commit. 
Other than that, LGTM

Commit:
Updated classifier
added gridsearch to optimize the kernel's C and gamma parameters for the SVM
Added numpy to dependencies
